### PR TITLE
Update table for state store given v1.0.1 hotfix

### LINKED ISSUE #1271: [CONTENT] Document etag requirement for actor state store
**What content needs to be created or modified?**
Documents about actor state store.
Includes the table here too: https://docs.dapr.io/operations/components/setup-state-store/supported-state-stores/

**Describe the solution you'd like**
Documents about actors state store should clarify that state stores must also support etag.

**Where should the new material be placed?**
Same content, just updated.

**Additional context**
Wait until this PR is merged, so the new behavior is actual: https://github.com/dapr/dapr/pull/2848

### PR DESCRIPTION
**Pull request by artursouza**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**

- [X] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [X] Commands include options for Linux, MacOS, and Windows within codetabs
- [ ] New file and folder names are globally unique
- [ ] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

Update table for state store given v1.0.1 hotfix.
MongoDB now supports eTag as of v1.0.1.
Actors state store now require to be transactional and support etag.

## Issue reference

Will close #1271

### CODE REVIEW

File: daprdocs/content/en/getting-started/install-dapr-selfhost.md [hunk 1]
```diff
@@ -52,8 +52,8 @@ dapr --version
 
 Output should look like this:
 ```
-CLI version: 1.0.0
-Runtime version: 1.0.0
+CLI version: 1.0.1
```
Comment (mukundansundar) [inline daprdocs/content/en/getting-started/install-dapr-selfhost.md]: was a new version released?
this should still be 1.0.0

File: daprdocs/content/en/operations/hosting/self-hosted/self-hosted-upgrade.md [hunk 2]
```diff
@@ -25,11 +25,11 @@ description: "Follow these steps to upgrade Dapr in self-hosted mode and ensure
...
 
-1. Ensure you are using the latest version of Dapr (1.0.0) with:
+1. Ensure you are using the latest version of Dapr (1.0.1) with:
 
    ```bash
    $ dapr --version
    
-   CLI version: 1.0.0
-   Runtime version: 1.0.0
+   CLI version: 1.0.1
```
Comment (mukundansundar) [inline daprdocs/content/en/operations/hosting/self-hosted/self-hosted-upgrade.md]: revert to 1.0.0 for CLI

**Review by mukundansundar — CHANGES_REQUESTED** [review]

File: daprdocs/content/en/operations/hosting/self-hosted/self-hosted-upgrade.md (hunk 2 above)
Comment (artursouza) [reply daprdocs/content/en/operations/hosting/self-hosted/self-hosted-upgrade.md]: Yep. Thanks.

File: daprdocs/content/en/getting-started/install-dapr-selfhost.md (hunk 1 above)
Comment (artursouza) [reply daprdocs/content/en/getting-started/install-dapr-selfhost.md]: Find replace error. Fixed. Thanks.

**Review by mukundansundar — APPROVED** [review]

**Review by orizohar — APPROVED** [review]

**Pull request merged**